uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/WorkPattern.php

```diff
@@ -45,6 +45,10 @@ public static function create($params) {
       $params['weight'] = self::getMaxWeight() + 1;
     }
 
+    if(!empty($params['id']) && isset($params['is_active']) && !$params['is_active']) {
```

Comment:
also in this case , I think you can just do this :

if (!empty($params['is_active']))

instead of : 

`if (isset($params['is_active']) && !$params['is_active'] )`